lib/client/db/postgres/mcp/mcp.go

```diff
@@ -0,0 +1,253 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+	"log/slog"
+	"net"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/mark3labs/mcp-go/mcp"
+
+	dbmcp "github.com/gravitational/teleport/lib/client/db/mcp"
+	"github.com/gravitational/teleport/lib/defaults"
+)
+
+// queryTool is the run query MCP tool definition.
+var queryTool = mcp.NewTool(dbmcp.ToolName(defaults.ProtocolPostgres, "query"),
+	mcp.WithDescription("Execute SQL query against PostgreSQL database connected using Teleport"),
+	mcp.WithString(queryToolDatabaseParam,
+		mcp.Required(),
+		mcp.Description("Teleport database resource URI where the query will be executed"),
+	),
+	mcp.WithString(queryToolQueryParam,
+		mcp.Required(),
+		mcp.Description("PostgresSQL SQL query to execute"),
+	),
+)
+
+type database struct {
+	*dbmcp.Database
+	pool *pgxpool.Pool
+}
+
+// Server PostgreSQL MCP server.
+type Server struct {
+	logger    *slog.Logger
+	databases map[string]*database
+}
+
+// NewServer initializes a PostgreSQL MCP server.
+func NewServer(ctx context.Context, cfg *dbmcp.NewServerConfig) (dbmcp.Server, error) {
+	s := &Server{logger: cfg.Logger, databases: make(map[string]*database)}
+
+	for _, db := range cfg.Databases {
+		if db.DatabaseUser == "" || db.DatabaseName == "" {
+			return nil, trace.BadParameter("must specify the username or database name used to connect to the database")
```

Comment:
The messages says "or" but the if statement above requires both to be set. Which one is it?